neurom/io/datawrapper.py

```diff
@@ -168,31 +177,30 @@ def _extract_sections(data_block):
             curr_section.ids.append(row_id)
 
         if row_id in sec_end_pts:
-            parent_section[curr_section.ids[-1]] = len(_sections) - 1
-            _sections.append(Section())
-            curr_section = _sections[-1]
+            parent_section[curr_section.ids[-1]] = len(sections) - 1
             # Parent-child discontinuity section
             if gap:
+                sections.append(DataBlockSection())
+                curr_section = sections[-1]
                 curr_section.ids.extend((parent_id, row_id))
                 curr_section.ntype = int(row[COLS.TYPE])
-                _gap_sections.add(len(_sections) - 2)
+                gap_sections.add(len(sections) - 2)
+            elif row_id != len(data_block) - 1:
+                # avoid creating an extra DataBlockSection for last row if it's a leaf
+                sections.append(DataBlockSection())
```

Comment:
I thought about that when writing it.  I think I'm going to rewrite some of this routine and its dependencies in a more 'numpy' way, so I'll do it then, if that's ok.